### PR TITLE
create sendVerificanionEmail feature when a principle user registers,…

### LINKED ISSUE #7: Log-In/Sign-Up: Vertify Account
During registration, a verification email will be sent to the user and they can confirm their registration by clicking the link in the verification email. Clicking the link takes them to the Sign-Up Form to continue the sign-up/registration process.

### PR DESCRIPTION
**Pull request by AikaAsan**
… backend codesplit into controllers and services

### CODE REVIEW

File: backend/service/principle-service.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const activate = async (activationLink) => {
+    const principle = await Principle.findOne({ activationLink });
+    console.log(principle);
+
+    if (!principle) {
+        throw new Error("the activation link is not correct");
+    }
+
+    principle.isActivated = true;
+
```
Comment (i-iurchuk) [inline backend/service/principle-service.js:34]: It is  necessary to check whether the account activation time has not expired. Sorry, I don't remember if this is part of the challenge. Maybe we need to create a separate task for this.

**Comment by i-iurchuk** [comment]
It is necessary to check whether the account activation time has not expired. Sorry, I don't remember if this is part of this task. Maybe we need to create a separate task for this.

**Comment by i-iurchuk** [comment]
Also, I don't know if we can verify somehow that the message was successfully sent.... :woman-shrugging::skin-tone-3:

File: backend/routes/index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const memberController = require("../controllers/member-controller");
+const invitationController = require("../controllers/invitation-controller");
+const familyController = require("../controllers/family-controller");
+
+router.post("/register", principleController.registration);
+router.get("/activate/:link", principleController.activate);
+router.post("/login", principleController.login);
+router.post("/member", memberController.memberRegistration);
+router.post("/invitation", invitationController.invitation);
+/**
```
[the remark this answers is not in the document]
Comment (AikaAsan) [reply backend/routes/index.js]: removed duplicated initializations. Wanted to put them in one index file but it didnt work, so left it initialized separately.

File: backend/routes/index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const express = require("express");
+const router = express.Router();
+const principleController = require("../controllers/principle-controller");
+const memberController = require("../controllers/member-controller");
+const invitationController = require("../controllers/invitation-controller");
+const familyController = require("../controllers/family-controller");
+
+router.post("/register", principleController.registration);
+router.get("/activate/:link", principleController.activate);
```
[the remark this answers is not in the document]
Comment (AikaAsan) [reply backend/routes/index.js]: upon registration an activation link (random unique string is created) and is appended to the uri ${process.env.API_URL}/api/activate/${activationLink} . It's a kind of verification code that is sent to a user and saved in the DB. By this verification code we can find an account that needs to be activated

File: backend/service/principle-service.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        // res.status(401).send({ message: "This user already exist" });
+    } else {
+        const activationLink = uuid.v4();
+        const p1 = new Principle({ email, password, activationLink });
+        await p1.save();
+        console.log("email:", email);
+        await mailService.sendActivationEmail(
+            email,
+            `${process.env.API_URL}/api/activate/${activationLink}`
+        );
```
[the remark this answers is not in the document]
Comment (AikaAsan) [reply backend/service/principle-service.js:18]: In the sprint google sheet I created a tab with env variables that have to added in the .env file locally. I cant add them to env.example file as it has sensitive information about the KIDS-FIRST email account

**Pull request closed without merging**